refinery/core/api.py

```diff
@@ -678,24 +671,34 @@ def filter_by_group(self, request, obj_list):
         return obj_list
 
     def obj_get(self, bundle, **kwargs):
-        dataset = SharableResourceAPIInterface.obj_get(self, bundle, **kwargs)
-        return self.handle_incomplete_dataset(dataset)
+        try:
+            dataset = DataSet.objects.get(uuid=kwargs["uuid"])
```

Comment:
I think we are avoiding the super call to ModelResource because of the wackyness mentioned in ShareableResourceApiInterfaces `obj_get()`

Regardless, calling `obj_get()` on an invalid dataset bundle is what leads to the `AttributeErrors` we've seen in the dehydrate method in the first place